lib/abilities.ex

```diff
@@ -9,5 +9,25 @@ defimpl Canada.Can, for: Pairmotron.User do
   def can?(%User{id: user_id}, action, %Group{owner_id: user_id})
     when action in [:edit, :update, :delete], do: true
 
+  def can?(%User{id: user_id}, :show, project = %Project{}) do
+    if is_nil(project.group_id) do
+      false
+    else
+      project = project |> Pairmotron.Repo.preload([{:group, :users}])
+      project.group.users
+      |> Enum.any?(fn user -> user.id == user_id end)
+    end
+  end
```

Comment:
Yep, totally possible. I didn't do it because you can't access struct properties in guard clauses, but forgot that I can just pattern match on the property directly. Not the first time I've forgotten that.

So since a retro is tied to a specific user, I think that it's probably okay to let someone edit all of their own retros, pretty much always. That's definitely something to think about though.